### DISCUSSION TITLE
Car Configurations

### DISCUSSION THREAD

**Discussion by tisoft** [SAIC-iSmart-API/documentation#4] [General]
I'd like to collect different car configurations, so that we can perhaps see what kind of variations we have in the API behavior.

If you use the java MQTT API, you can find it in the topic `saic/EMAIL/vehicles/VIN/_internal/configuration/raw`.

This does not contain any personal data.

MG5 Short Range Luxury:
```name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:1;name:ENERGY,code:ENERGY,value:0;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:0;name:Battery,code:BType,value:2```

Just paste your configuration, together with the car Model. I will collect this information and create a nice table overview here: https://github.com/SAIC-iSmart-API/documentation/blob/main/docs/examples/v1_1/501_513.adoc

**Comment by tosate** [comment]
MG ZS EV, Standard Range, Luxury
`name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:0;name:AirbagS,code:Q09,value:0;name:Sunroof,code:S35,value:3;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:1;name:ENERGY,code:ENERGY,value:1;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:1;name:Battery,code:BType,value:2`

**Comment by DennisGeorge77** [comment]
MG4 Standard
`name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:0;name:ENERGY,code:ENERGY,value:1;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:1;name:Battery,code:BType,value:2`

**Reply by tisoft** [reply]
This is a MG4 SR Comfort, right? I have seen a MG4 with HeatedSeat: 1 somewhere, which must be a Luxury then.

**Reply by DennisGeorge77** [reply]
No. It is a MG4 Standard.

**Reply by tisoft** [reply]
Ah, my bad... mg4 has different model names than mg5.

So if anyone has a comfort or luxury, I'd like to have the data. 😄

**Comment by c1olli** [comment]
MG 4 Comfort

`name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:0;name:ENERGY,code:ENERGY,value:1;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:1;name:Battery,code:BType,value:1`

**Comment by bigbenFE** [comment]
MG 4 Luxury

`name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:1;name:KEYPOS,code:KEYPOS,value:0;name:ENERGY,code:ENERGY,value:1;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:1;name:Battery,code:BType,value:1`

**Comment by pepelillo** [comment]
MG Marvel R Luxury
name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:0;name:AirbagS,code:Q09,value:0;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101100000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:1;name:ENERGY,code:ENERGY,value:1;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:Battery,code:BType,value:1

**Comment by treerunner1** [comment]
MG5 Long Comfort
name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:1;name:ENERGY,code:ENERGY,value:0;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:0;name:Battery,code:BType,value:1

**Comment by nanomad** [comment]
MG4 Luxury with R24

`name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:1;name:KEYPOS,code:KEYPOS,value:0;name:ENERGY,code:ENERGY,value:1;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:1;name:Battery,code:BType,value:1
`

**Comment by klefreak** [comment]
MG5 Long Range Luxury

`name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:1;name:ENERGY,code:ENERGY,value:0;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:0;name:Battery,code:BType,value:1`

**Comment by bisscuitt** [comment]
MG ZS EV Exceed (Australia):
`name:Tire pressure monitoring system,code:J17,value:1;name:Regular airbags,code:Q00,value:1;name:Front-seat airbags,code:Q01,value:0;name:Airbag switch,code:Q09,value:0;name:Sun Roof,code:S35,value:0;name:Remote control,code:S61,value:1;name:Air conditioning,code:T11,value:1;name:Electric Power Steering,code:EPS,value:1;name:Security alert,code:SA64,value:0111110000000000000010000100100100000010100000000000000000000110;name:Bonnut Status,code:BONNUT,value:1;name:Door Status,code:DOOR,value:1111;name:Boot Status,code:BOOT,value:1;name:Engine Status,code:ENGINE,value:1;name:Electric Vehicle,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:Key Position,code:KEYPOS,value:1;name:Energy state,code:ENERGY,value:1;name:Battery Voltage,code:BATTERY,value:1;name:Interior Temperature,code:INTEMP,value:1;name:Exterior Temperature,code:EXTEMP,value:1;name:Window Status,code:WINDOW,value:0000;name:Left-Right Driving,code:LRD,value:1;name:BLE CarKey,code:BTKEY,value:0;name:Battery Type,code:BTYPE,value:2;`

**Reply by bisscuitt** [reply]
Note the trailing `;` which others do not seem to have.

**Reply by mschmiedel** [reply]
Is the trailing ; really in the topic, or might it be the way it was extracted?

**Comment by mschmiedel** [comment]
MG ZS EV SR Luxury

name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:0;name:AirbagS,code:Q09,value:0;name:Sunroof,code:S35,value:3;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:1;name:ENERGY,code:ENERGY,value:1;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:1;name:Battery,code:BType,value:2

**Comment by Maart3nL** [comment]
MG4 Luxury (64KWh Dutch EU Model)

`name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:1;name:KEYPOS,code:KEYPOS,value:0;name:ENERGY,code:ENERGY,value:1;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:1;name:Battery,code:BType,value:1`

**Comment by xtreamjava** [comment]
MG ZS EV (MY 2021) Essence Facelift AUSTRALIA

name:Tire pressure monitoring system,code:J17,value:1;name:Regular airbags,code:Q00,value:1;name:Front-seat airbags,code:Q01,value:0;name:Airbag switch,code:Q09,value:0;name:Sun Roof,code:S35,value:0;name:Remote control,code:S61,value:1;name:Air conditioning,code:T11,value:1;name:Electric Power Steering,code:EPS,value:1;name:Security alert,code:SA64,value:0111110000000000000010000100100100000010100000000000000000000110;name:Bonnut Status,code:BONNUT,value:1;name:Door Status,code:DOOR,value:1111;name:Boot Status,code:BOOT,value:1;name:Engine Status,code:ENGINE,value:1;name:Electric Vehicle,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:Key Position,code:KEYPOS,value:1;name:Energy state,code:ENERGY,value:1;name:Battery Voltage,code:BATTERY,value:1;name:Interior Temperature,code:INTEMP,value:1;name:Exterior Temperature,code:EXTEMP,value:1;name:Window Status,code:WINDOW,value:0000;name:Left-Right Driving,code:LRD,value:1;name:BLE CarKey,code:BTKEY,value:0;name:Battery Type,code:BTYPE,value:2;

**Comment by cybersmart-eu** [comment]
MG5 Short Range Comfort (Germany, MY 2021)
`name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:1;name:ENERGY,code:ENERGY,value:0;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:0;name:Battery,code:BType,value:2`

**Reply by bjoernkowarsch** [reply]
MG5 Long Range Luxury
name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:0;name:KEYPOS,code:KEYPOS,value:1;name:ENERGY,code:ENERGY,value:0;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:1;name:WINDOW,code:WINDOW,value:0000;name:LRD,code:LRD,value:0;name:BTKEY,code:BTKEY,value:0;name:Battery,code:BType,value:1

**Comment by townsmcp** [comment]
**MG S6 EV (MIS3E) — Trophy Long Range (UK, MY2025)**

`name:Tire,code:J17,value:1;name:Airbag,code:Q00,value:1;name:FAirbag,code:Q01,value:1;name:AirbagS,code:Q09,value:1;name:Sunroof,code:S35,value:0;name:Remote control,code:S61,value:1;name:AC,code:T11,value:1;name:EPS,code:EPS,value:1;name:SA64,code:SA64,value:0111110000000000001000000100101000000010100000000000000000000110;name:BONNUT,code:BONNUT,value:1;name:DOOR,code:DOOR,value:1111;name:BOOT,code:BOOT,value:1;name:ENGINE,code:ENGINE,value:1;name:EV,code:EV,value:0;name:HeatedSeat,code:HeatedSeat,value:1;name:KEYPOS,code:KEYPOS,value:1;name:ENERGY,code:ENERGY,value:1;name:BATTERY,code:BATTERY,value:1;name:INTEMP,code:INTEMP,value:1;name:EXTEMP,code:EXTEMP,value:0;name:WINDOW,code:WINDOW,value:1111;name:LRD,code:LRD,value:1;name:BTKEY,code:BTKEY,value:2;name:Battery,code:BType,value:1;name:SBH,code:SBH,value:1;name:TR,code:TR,value:1;name:BHI,code:BHI,value:1;name:HeatedSeat2,code:HEATEDSEAT2,value:2;name:SWH,code:SWH,value:1;name:BTKNOTIFY,code:BTKNOTIFY,value:1;name:SS,code:SS,value:0;name:SC,code:SC,value:5`

A few things that look new compared to the models already collected here:

- **`WINDOW` reports `1111`, not `0000`.** Every earlier car in this thread shows `WINDOW=0000`; this newer model appears to actually populate it. (Worth noting: `EV=0` even though this is a full EV, so that field still isn't a reliable EV indicator.)
- Several **codes I haven't seen documented here yet**: `SBH`, `TR`, `BHI`, `HEATEDSEAT2` (=2), `SWH`, `BTKNOTIFY`, `SS`, `SC`.
- `HEATEDSEAT2` and `SWH` are interesting — this car has front *and* rear heated seats plus a heated steering wheel, so I suspect `HeatedSeat`=front, `HEATEDSEAT2`=rear (second row), and `SWH`=steering wheel heat. From decrypting the app's own commands, the rear seats and steering wheel are controlled via their own request types (rear seats use the heated-seats request with per-seat param IDs; steering wheel uses a request type not currently in the client library). Happy to share more detail if useful.

Note `EV=0` and `EXTEMP=0` here, which differ from most other entries.
